File: Cycle_Rental_System/src/main/java/com/springproject/Cycle_Rental_System/services/admin/AdminService.java
package com.springproject.Cycle_Rental_System.services.admin;

import java.io.IOException;
import java.util.List;

import com.springproject.Cycle_Rental_System.dto.CycleDto;

public interface AdminService {
	boolean postCycle(CycleDto cycleDto);
	List<CycleDto> getAllCycles();
	void deleteCycle(Long id);
	CycleDto getCycleById(Long id);
	boolean updateCycle(Long cycleId, CycleDto cycleDto) throws IOException;

}

File: cycle_rental_app_angular/src/app/modules/admin/services/admin.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

const BASIC_URL = ["http://localhost:8081"];

@Injectable({
  providedIn: 'root'
})
export class AdminService {
  constructor(private http: HttpClient) {}

  postCycle(cycleDto: any): Observable<any> {
    return this.http.post(BASIC_URL + "/api/admin/cycle", cycleDto, {
      headers: this.createAuthorizationHeader()
    });
  }
  getAllCycles(): Observable<any> {
    return this.http.get(BASIC_URL + "/api/admin/cycles", {
      headers: this.createAuthorizationHeader()
    });
  }

  deleteCycle(id: number): Observable<any> {
    return this.http.delete(BASIC_URL + "/api/admin/cycle/" + id, {
      headers: this.createAuthorizationHeader()
    });
  }

  getCarById(id: number): Observable<any> {
    return this.http.get(BASIC_URL + "/api/admin/car/" + id, {
      headers: this.createAuthorizationHeader()
    });
  }

  createAuthorizationHeader(): HttpHeaders {
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + StorageService.getToken()
    );
  }
}